### PR TITLE
Use ChatGPT to write our recipe pitches!

### PR DESCRIPTION
**Pull request by mbenfowler**
## Description
This pr will integrate a ChatGPT fetched recipe pitch when hovering over each recipe to entice the user to click!
Right now, we have a slide-down appear when a user hovers over a recipe on our recipes view, but all it says is "Read more..." for all of them.  So instead of drafting and hard-coding descriptions for each of them, why not put AI to work for us here!

The value of this proposal is two-fold:
- A deeper understanding of fetch and live APIs, ones with more complex request structures including auth
- Utilizing a technology (AI) that is rapidly expanding and will likely soon be playing an every day role in our lives and the lives of our future users/customers.

**Constraints:**
- Only use fetch functionality that we've learned up to this point
- No need to install further chatgpt api wrapper packages
- This function would live in our apiCalls with the others, and it would take our fetched recipes object as an argument where we add a new pitch property to each recipe with the response text.

Screenshot 2023-05-31 at 1 29 46 PM

## Testing Notes
- Observe new spinner as we wait the arbitrary 2 seconds for our pitch data
  - Make sure spinner is hidden once the 2 seconds elapse
- Hover over recipe cards to see AI created pitches!
  - text should scroll up and stop once the text block end is reached
  - text scroll animation restarts if you stop hovering and re-hover over a recipe
- If data isn't received in time, the pitch will default to "Click to read more..."
  - You can comment out the call to openai to simulate this
- `config.js` created locally to hold our secret api key and `.gitignore` updated so that we don't push that information up to our public repo.  This means we will currently only be able to get our ai created pitches locally.

### CODE REVIEW

**Comment by mbenfowler** [comment]
[[SUPER EXTENSION] Enable ChatGPT to write 10-word pitches for our recipe hover-overs](https://trello.com/c/5hvb6wXp/14-super-extension-enable-chatgpt-to-write-10-word-pitches-for-our-recipe-hover-overs)

File: src/apiCalls.js [hunk 1]
```diff
@@ -1,8 +1,9 @@
 //IMPORTS 
 import { getRandomUser } from "./users";
-import { pageLoadRenders } from "./domUpdates";
+import { pageLoadRenders, hideSpinner } from "./domUpdates";
```
Comment (taranveersingh93) [inline src/apiCalls.js:3]: nice

File: src/styles.css [hunk 2]
```diff
@@ -308,10 +308,46 @@ nav {
...
+  border-right-color: #0dd;
+  border-bottom-color: #f90;
+  animation: spinner .6s linear infinite;
+}
+
+@keyframes spinner {
+  to {
+      transform: rotate(360deg);
+  }
 }
```
Comment (lauraguerra1) [inline src/styles.css:352]: woooooooowza

File: src/apiCalls.js [hunk 3]
```diff
@@ -1,8 +1,9 @@
...
 import { copyItem } from "./helper-functions";
 import { populateTags, calculateRecipeCost, getIngredientAmounts, getInstructions } from './recipes';
+import { config } from "../config.js"
```
Comment (lauraguerra1) [inline src/apiCalls.js]: pls comment this out before merging, otherwise looks amazing

**Review by lauraguerra1 — CHANGES_REQUESTED** [review]

File: src/apiCalls.js (hunk 1 above)
Comment (taranveersingh93) [reply src/apiCalls.js:3]: Can we just have some padding on the hover-card or margin on the pitch text?
Otherwise it looks great!

**Review by taranveersingh93 — CHANGES_REQUESTED** [review]
Can we just have some padding on the hover-card or margin on the pitch text?
Otherwise it looks great!

File: src/apiCalls.js (hunk 1 above)
Comment (mbenfowler) [reply src/apiCalls.js:3]: great call, done!

File: src/apiCalls.js (hunk 3 above)
Comment (mbenfowler) [reply src/apiCalls.js]: done

**Review by lauraguerra1 — APPROVED** [review]

**Review by taranveersingh93 — APPROVED** [review]

**Pull request merged**